source/Octopus.Tentacle/Kubernetes/KubernetesScriptPodCreator.cs

```diff
@@ -225,17 +215,70 @@ await CreateScriptContainer(command, workspace, podName, scriptName, homeDir)
 
             var createdPod = await podService.Create(pod, cancellationToken);
             podMonitor.AddPendingPod(command.ScriptTicket, createdPod);
-
             LogVerboseToBothLogs($"Executing script in Kubernetes Pod '{podName}'.", tentacleScriptLog);
         }
 
+        async Task<IList<V1Container>> CreateInitContainers(StartKubernetesScriptCommandV1 command, string podName, string homeDir, string workspacePath)
+        {
+            if (!command.IsRawScript)
```

Comment:
Analogous to splitting the contract, this class here can be split into two separate implementations for the `IKubernetesScriptPodCreator`. This means that the non raw script pod creator wouldn't need to know about init containers and the code can be simpler.

Reply:
@scme0 Yeah, I think this is a good idea. if there is some shared logic, then we can have an `abstract` `KubernetesScriptPodCreator` that they both inherit from

Reply:
I've just made a new class `KubernetesRawScriptPodCreator` which inherits from `KubernetesScriptPodCreator` directly. I had to make a few methods `protected virtual` and then I could override them in the raw script pod creator to make it work as expected. Let me know what you think.